File: client/ng-mes-auto/src/app/models/form-field-config.ts
export class FormFieldConfig {
  static ALL_VALUE_TYPE = ['STRING', 'BOOLEAN', 'NUMBER'];
  static ALLL_INPUT_TYPE = ['DEFAULT', 'SELECTION'];
  id: string;
  name: string;
  required = true;
  multi = false;
  valueType = FormFieldConfig.ALL_VALUE_TYPE[0];
  inputType = FormFieldConfig.ALLL_INPUT_TYPE[0];
  selectOptions: string[];

  get isInputString(): boolean {
    return this.valueType === 'STRING' && this.inputType === 'DEFAULT';
  }

  get isInputNumber(): boolean {
    return this.valueType === 'NUMBER' && this.inputType === 'DEFAULT';
  }

  get isInputBoolean(): boolean {
    return this.valueType === 'BOOLEAN';
  }

  get isSelectString(): boolean {
    return this.valueType === 'STRING' && this.inputType === 'SELECTION';
  }

  get isSelectNumber(): boolean {
    return this.valueType === 'NUMBER' && this.inputType === 'SELECTION';
  }

  static assign(...sources: any[]): FormFieldConfig {
    const result = Object.assign(new FormFieldConfig(), ...sources);
    return result;
  }

  static hasSelectOptions(valueType: string, inputType: string): boolean {
    if (valueType === 'ARRAY' || inputType === 'SELECTION') {
      return true;
    }
    return false;
  }
}

File: mes-root/mes-auto-core/src/main/java/com/hengyi/japp/mes/auto/domain/data/FormFieldConfig.java
package com.hengyi.japp.mes.auto.domain.data;

import lombok.Data;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Collection;

/**
 * @author jzb 2018-10-25
 */
@Data
public class FormFieldConfig {
    private String id;
    @NotBlank
    private String name;
    @NotNull
    private FormFieldConfig.FormFieldValueType valueType;
    @NotNull
    private FormFieldConfig.FormFieldInputType inputType;
    private boolean required;
    private boolean multi;
    private Collection<String> selectOptions;

    /**
     * 值类型
     */
    public enum FormFieldValueType {
        STRING,
        BOOLEAN,
        NUMBER,
    }

    /**
     * 输入方式
     */
    public enum FormFieldInputType {
        DEFAULT,
        SELECTION,
    }
}
